appnexus/cursor.py

```diff
@@ -86,15 +75,17 @@ def get_page(self, start_element=0, num_elements=None):
         specs.update(start_element=start_element, num_elements=num_elements)
         return self.client.get(self.service_name, **specs)
 
-    def iter_pages(self, skip_elements=0):
+    def iter_pages(self):
         """Iterate as much as needed to get all available pages"""
-        start_element = skip_elements
+        start_element = self._skip
+        num_elements = min(self._limit, self.batch_size)
         count = -1
         while start_element < count or count == -1:
-            page = self.get_page(start_element)
+            page = self.get_page(start_element, num_elements)
             yield page
-            start_element = page["start_element"] + page["num_elements"]
-            count = page["count"]
+            start_element = start_element + page["num_elements"]
+            num_elements = min(page["count"] - num_elements, self.batch_size)
+            count = min(page["count"], self._skip + self._limit)
```

Comment:
Can't we just keep `num_elements`, and entirely remove `count` from the method?

Reply:
I didn't address this comment.
I remember trying hard to handle all corners cases and also avoiding all these intermediate variables and always bumping into a problem or having a pretty unreadable code.
I think the code is both functional and maintenable as is so I would keep it that way.